presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -522,10 +528,29 @@ private synchronized void updateFinalTaskInfo(TaskInfo finalTaskInfo)
     private synchronized void checkAllTaskFinal()
     {
         if (stateMachine.getState().isDone() && tasksWithFinalInfo.containsAll(allTasks)) {
+            if (getFragment().getStageExecutionDescriptor().isStageGroupedExecution()) {
```

Comment:
It feels like this sanity check is an overkill. But feel free to leave it if you feel strongly about it.

Reply:
The original purpose is to make sure all tests passed :). I agree, maybe a simple check state that `finishedLifespans.size() <= totalLifespans`

Reply:
It just that this piece of code seems to be too big to be a simple sanity check, and it creates an impression as if it does something more than that. Anyhow, feel free to keep =) I don't have a strong opinion here. Perhaps i should've marked this comment as a "nit" from the beginning.